src/wast-parser.cc

```diff
@@ -3439,7 +3439,11 @@ Result WastParser::ParseModuleCommand(Script* script, CommandPtr* out_command) {
                 error.loc.offset, error.message.c_str());
         }
       }
-      *module = std::move(*m.get());
+      if (m) {
+        *module = std::move(*m.get());
```

Comment:
Is it possible for `ParseWatModule` above to succeed without `*m` being set?

Would this work instead:

```
if (errors.length()) { 
  return Result::Error;
}
assert(*m);
```

Reply:
doesn't `ParseWatModule` also return a `Result`? probably a good idea to check that instead.

Reply:
Yes that sounds good.    You can do something like this

```
if (Failed(ParseWatModule(lexer.get(), &m, &errors, options_)) { 
   for (const auto& error : errors) {
      ...
   }
   return Result::Error;
}
assert(errors.length == 0);
```